### PR TITLE
book-store-impl-glexception

### PR DESCRIPTION
**Pull request by SkillfulDev**

### CODE REVIEW

File: src/main/java/ua/chernonog/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    @ExceptionHandler(SQLIntegrityConstraintViolationException.class)
+    public ResponseEntity<Object> handleSqlIntegrityConstraintViolationException(
+            SQLIntegrityConstraintViolationException e) {
+        return getStandartTemplateOfResponseEntity(e);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException e) {
+        return getStandartTemplateOfResponseEntity(e);
```
Comment (andrii-hoienko) [inline src/main/java/ua/chernonog/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: Is this case getStandartTemplateOfResponseEntity method will return HttpStatus.CONFLICT, but I believe HttpStatus.NOT_FOUND should be returned, please check other methods as well and choose appropriate status code

File: src/main/java/ua/chernonog/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    private ResponseEntity<Object> getStandartTemplateOfResponseEntity(
+            Throwable e) {
+        body.put(TIMESTAMP, LocalDateTime.now());
+        body.put(STATUS, HttpStatus.BAD_REQUEST);
+        body.put(ERRORS, e.getMessage());
+        return new ResponseEntity<>(
+                body, HttpStatus.CONFLICT);
```
Comment (andrii-hoienko) [inline src/main/java/ua/chernonog/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: Bad request status not matches with conflict status, please be consistent or remove body.put(STATUS, HttpStatus.BAD_REQUEST);

File: src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
 
 @Data
 public class CreateBookRequestDto {
     private Long id;
+    @NotEmpty(message = "Title cannot be empty")
     private String title;
+    @NotEmpty(message = "Author cannot be empty")
     private String author;
+    @NotEmpty(message = "ISBN cannot be empty")
+    @Pattern(regexp = "^(?:\\d{9}[\\dX]|97[89]\\d{10})$", message = "Invalid ISBN format")
```
Comment (andrii-hoienko) [inline src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java]: I believe that there are already exists some @ISDN validation annotation, please check it out

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, but some improvements required

File: src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,15 +1,30 @@
...
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.validator.constraints.ISBN;
+import org.hibernate.validator.constraints.URL;
 
 @Data
 public class CreateBookRequestDto {
     private Long id;
+    @NotEmpty(message = "Title cannot be empty")
```
Comment (yevheniisynytsia) [inline src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java]: do you want to accept `" "` values?
I think `@NotBlank` is better 
https://www.baeldung.com/java-bean-validation-not-null-empty-blank

File: src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,15 +1,30 @@
...
     private String title;
+    @NotEmpty(message = "Author cannot be empty")
     private String author;
+    @ISBN(message = "Invalid ISBN format")
+    @NotEmpty(message = "ISBN cannot be empty")
```
Comment (yevheniisynytsia) [inline src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java]: I think `@NotNull` is enough here, not sure how `@ISBN` works in details though

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]

File: src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java (hunk 5 above)
Comment (SkillfulDev) [reply src/main/java/ua/chernonog/onlinebookstore/dto/request/CreateBookRequestDto.java]: Using both annotations together ensures that the field is checked for void first and then against a specific format. They do not contradict each other, but complement the validation process, ensuring that the ISBN field is not empty, and correctly formatted. Changed just @NotNull to @NotBlank

**Review by yevheniisynytsia — APPROVED** [review]
Well done!

**Pull request closed without merging**